src/hardware/openbench-logic-sniffer/protocol.c

```diff
@@ -73,14 +73,28 @@ static int ols_send_longdata(struct sr_serial_dev_inst *serial, uint8_t command,
 
 SR_PRIV int ols_send_reset(struct sr_serial_dev_inst *serial)
 {
-	unsigned int i;
+	int i, ret, delay_ms;
+	char dummy[16];
+
+	/* Drain all data so that the remote side is surely listening. */
+	while ((ret = serial_read_nonblocking(serial, &dummy, 16)) > 0);
```

Comment:
I changed the formatting, on both instances to make it more obvious.

Yes, it might take a while to dump the data. But we need it because we need the device to be in the state we think it is: If it's still dumping data from before we reset it, we're not expecting this after the reset and the whole communication afterwards is messed up. Thus, I introduce this period of quiescence to re-sync the device and the driver.

(One might argue whether we need to drain data before issuing the reset, though. But I don't remember the protocol in detail anymore to make a statement on this)